apps/condo/domains/banking/utils/serverSchema/BankCategory.js

```diff
@@ -0,0 +1,40 @@
+const { BankCategory } = require('./index')
+const { DV_SENDER_FOR_NESTED_SET_UTIL } = require('../../constants')
+
+/**
+ * Sets indices for a subtree of specified root node using Depth-First Search
+ *
+ * Call this function manually after you build or changed a tree of `BankCategory` records.
+ *
+ * NOTE(antonal): This function is impossible to use in `afterChange` hook of a `BankCategory` model,
+ * to get, at first sight, an auto updating feature after moving a category as a child to another.
+ * Because update queries for `BankCategory` can be executed in unknown time and order,
+ * result will be unpredictable and greedy on performance.
+ *
+ * @param context
+ * @param category - `BankCategory` node that needs to be recursively traversed
+ * @param i - starting index for root node
+ * @param depth - level index within a tree
+ * @return {Promise<number>} - right index after recursive traversing of specified `BankCategory` node
+ */
+async function buildNestedSetIndex (context, category, i = 1, depth = 0) {
```

Comment:
The function should be executed every time when `parent` field was updated.

Reply:
I have written a big comment about usage of this function and why an automatic invocation will cause problems.